server/model/monitor.js

```diff
@@ -395,19 +397,28 @@ class Monitor extends BeanModel {
                             }
                             const lastBeat = await Monitor.getPreviousHeartbeat(child.id);
 
-                            // Only change state if the monitor is in worse conditions then the ones before
+                            // Only change state if the monitor is in worse conditions than the ones before
                             // lastBeat.status could be null
                             if (!lastBeat) {
                                 bean.status = PENDING;
                             } else if (bean.status === UP && (lastBeat.status === PENDING || lastBeat.status === DOWN)) {
                                 bean.status = lastBeat.status;
+                                downChildName = child.name;
                             } else if (bean.status === PENDING && lastBeat.status === DOWN) {
                                 bean.status = lastBeat.status;
+                                downChildName = child.name;
+                            }
+                            if (lastBeat.status === DOWN) {
+                                failureCount++;
                             }
                         }
 
                         if (bean.status !== UP) {
-                            bean.msg = "Child inaccessible";
+                            if (failureCount <= 1) {
+                                bean.msg = `Child '${downChildName}' inaccessible`;
+                            } else {
+                                bean.msg = `Child '${downChildName}' and ${failureCount - 1} others inaccessible`;
+                            }
```

Comment:
```suggestion
                            if (downChildNames.lengh === 1) {
                                bean.msg = `Child '${downChildName[0]}' inaccessible`;
                            } else {
                                bean.msg = `Child '${downChildNames[0]}' and ${downChildNames.lengh - 1} others inaccessible`;
                            }
```

Reply:
This is nothing short of ridiculous. Collect a list of names, just to get the first name from the list.

And you've misspelled length. Your code wouldn't even work. I'll just keep my own private fork of this project, and shape it to my own needs. I have many enhancements in mind, and working on this trivial already-existing issue was just my way of testing the waters to see if it is worth contributing to the project. People like you make open source worse than working for a corporation with bad management.

Good riddance.

Reply:
This is nothing short of ridiculous. Collect a list of names, just to get the first name from the list.  
  
And you've misspelled length. Your code wouldn't even work. I'll just keep my own private fork of this project, and shape it to my own needs. I have many enhancements in mind, and working on this trivial already-existing issue was just my way of testing the waters to see if it is worth contributing to the project. People like you make open source worse than working for a corporation with bad management.

 Good riddance.

Reply:
@louislam Do you have an opinion?

Reply:
> Collect a list of names, just to get the first name from the list.

first name and the lenght of the list.

> you've misspelled length

fixed it

> I'll just keep my own private fork of this project, and shape it to my own needs

That is fine by us.
Forking a project doesn’t have to be a bad thing. Being able to copy and modify projects is one of the best things about open source[^1]

> People like you make open source worse than working for a corporation with bad management.

Sorry that you feel this way. If there is something concrete that I can change in the way I review work, that would be appreciated (I am not even a junior dev in terms of job experience right now)

[^1]: https://opensource.guide/best-practices/#let-others-build-the-solutions-they-need